contracts/Market.sol

```diff
@@ -292,59 +287,84 @@ contract Market is IMarket, Ownable, ERC721 {
 		// add to the total potential payout for this proposition
 		_potentialPayout[propositionId] += payout;
 
+		uint64 index = _getCount();
+
 		_bets.push(
 			Bet(propositionId, marketId, wager, payout, end, false, _msgSender())
 		);
 
-		// use _getCount() to avoid stack too deep
-		_marketBets[marketId].push(_getCount());
-		_mint(_msgSender(), _getCount() - 1);
+		_marketBets[marketId].push(index);
+		_mint(_msgSender(), index);
```

Comment:
why not `msg.sender`?

Reply:
To stop sneakyness https://forum.openzeppelin.com/t/why-use-msgsender-rather-than-msg-sender/4370